plugins/elanfp/fu-elanfp-device.c

```diff
@@ -90,10 +127,22 @@ static gboolean
 fu_elanfp_iap_recv_status(FuElanfpDevice *self, guint8 *buf, gsize bufsz, GError **error)
 {
 	GUsbDevice *usb_device = fu_usb_device_get_dev(FU_USB_DEVICE(self));
+	guint16 pid = (guint)fu_usb_device_get_pid(FU_USB_DEVICE(self));
+	guint8 endpoint = ELAN_EP_CMD_IN;
 	gsize actual = 0;
+	gboolean is_new_protocol = FALSE;
+
+
+	if ((pid != 0x0C7E) && (pid != 0x0C82)) {
+					is_new_protocol = TRUE;
```

Comment:
what does "new" mean? When the next protocol version is added will that be new_new? or new2? :) I think you want to use fu_device_register_private_flag() on a new flag called "use-bulk-transfer" and then set this in the quirk file, i.e.

    [USB\VID_04F3&PID_0C82]
    Flags = use-bulk-transfer

and then use `fu_device_has_private_flag` in the code.

Reply:
Fixed in commit id e1fc9869b5b22a463de2ecc01a333aecad900d44.